### DISCUSSION TITLE
How can I write xlsx file to cloud using stream?

### DISCUSSION THREAD

**Discussion by kyungyeon-byun-imweb-me** [exceljs/exceljs#1714] [Q&A] [answered]
I want to upload to the cloud using a stream.

example..

```javascript
const ExcelJS = require('exceljs');
const excel = new ExcelJS.stream.xlsx.WorkbookWriter(cloudStream);
```

Is this impossible without writing and reading files locally?

**Comment by legopin** [comment, marked as answer]
It's possible, you just need to get a writableStream to your cloud storage provider (Amazon S3 or Google GCS)

The documentation below gives some idea about how to use the streaming mode
https://github.com/exceljs/exceljs#streaming-xlsx
